src/jackdaw/streams/extras.clj

```diff
@@ -3,13 +3,117 @@
   {:license "BSD 3-Clause License <https://github.com/FundingCircle/jackdaw/blob/master/LICENSE>"}
   (:require [clojure.tools.logging :as log]
             [jackdaw.streams :as js]
+            [jackdaw.streams.interop :as jsi]
+            [jackdaw.streams.protocols :as jsp]
             [clojure.spec.alpha :as s])
-  (:import org.apache.kafka.common.TopicPartition
-           org.apache.kafka.streams.processor.StateRestoreListener
+  (:import (jackdaw.streams.interop CljKStream CljKTable CljGlobalKTable CljKGroupedTable
+                                    CljKGroupedStream CljTimeWindowedKStream CljSessionWindowedKStream)
+           org.apache.kafka.common.TopicPartition
+           (org.apache.kafka.streams StreamsBuilder)
+           (org.apache.kafka.streams.kstream KStream KTable KGroupedStream KGroupedTable
+                                             GlobalKTable TimeWindowedKStream
+                                             SessionWindowedKStream Transformer ValueTransformer)
+           (org.apache.kafka.streams.processor ProcessorContext StateRestoreListener)
+           (org.apache.kafka.streams.state Stores)
            [java.time Instant Duration]))
 
 (set! *warn-on-reflection* true)
 
+(defn transformer-with-ctx
```

Comment:
Hi :-)

Do we care about consistency with how the processor function is dealt with?
- this is now not in the same namespace so perhaps awkward to find
- processor natively sorts the context out for you but transform doesn't - is backwards compatibility a consideration here?  Can we introduce a breaking change for consistent behaviour?  Or call it `transfrom!` maybe?
- the implementation with reify in this change is different to the implementation for processor which we based our changes on see `FnTransformer` and `transformer` here https://github.com/FundingCircle/originations-application-tracker/blob/application-creation-command-handler/src/tracker/stream_utils.clj which was based on https://github.com/FundingCircle/jackdaw/blob/30b341d74c42dc0049221648113ac43acd572115/src/jackdaw/streams/lambdas.clj#L145

Reply:
did some moving around which resolves these (very valid) points hopefully